packages/prepare-commit-msg/src/utils/convert.ts

```diff
@@ -0,0 +1,61 @@
+import { type Author, coauthors } from "@coauthors/core";
+import { debug } from "./log";
+
+const COAUTHORS_PREFIX = "coauthors:";
+
+interface MessageInfo {
+  result: string;
+  isConverted: boolean;
+}
+
+export const convert = async (message: string): Promise<MessageInfo> => {
+  debug(`Original commit message: ${message}`);
+
+  const lines = message.split("\n");
+
+  const coauthorsLineIndex = lines.findIndex((line) =>
+    line.startsWith(COAUTHORS_PREFIX),
+  );
+  const coauthorsLine = lines.find((line) =>
+    line.trimStart().startsWith(COAUTHORS_PREFIX),
+  );
+
+  const authors = coauthorsLine
+    ?.substring(COAUTHORS_PREFIX.length, coauthorsLine.length)
+    .split(",")
+    .map(parseAuthorWithParentheses);
+
+  const coauthorsResult = authors ? await coauthors(authors) : undefined;
+
+  if (coauthorsLineIndex >= 0 && coauthorsResult) {
+    lines[coauthorsLineIndex] = coauthorsResult;
+  }
+
+  return {
+    isConverted: !!coauthorsLine,
+    result: lines.join("\n"),
+  };
+};
+
+const parseAuthorWithParentheses = (inputString: string): Author => {
```

Comment:
```ts
export const parseAuthors = (args: string[]) =>
  args.map((arg) => {
    const [user, name] = arg.split(/\(([^)]+)\)/);
    return { user, name };
  });
```

I'm wondering why you created a new function that looks like it does the same thing, when there is a parseAuthors util in the cli that does it with regular expressions?

https://github.com/coauthors/coauthors/blob/main/packages/cli/src/utils/parseAuthors.ts

Reply:
Then, you mean that I should use parseAuthors from @coauthors/cli? or any alternatives more specific suggestion is needed for me. I was worry about that using parseAuthors of @coauthors/cli should require @coauthors/prepare-commit-msg have @coauthors/cli as dependency. I don't want to make another unnecessary public interface of @coauthors/cli

Reply:
My opinion is to move the parseAuthors function to core. Anyway, this is likewise not important for this PR, so we'll discuss this later
https://github.com/coauthors/coauthors/issues/35